### PR TITLE
tools/rados: Improve list-inconsistent json format

### PR DESCRIPTION
**Pull request by dzafman**
This is on top of pull another pull request.

We should discuss what the output format looks like.

### CODE REVIEW

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -1317,20 +1325,26 @@ static void dump_inconsistent(const inconsistent_obj_t& inc,
...
+    f.dump_string("error", "attr_unexpected");
   if (inc.has_shard_missing())
-    f.dump_bool("missing", inc.has_shard_missing());
+    f.dump_string("error", "missing");
   if (inc.has_stat_error())
-    f.dump_bool("stat_err", inc.has_stat_error());
+    f.dump_string("error", "stat_err");
   if (inc.has_read_error())
-    f.dump_bool("read_err", inc.has_read_error());
+    f.dump_string("error", "read_err");
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: @dzafman could you take this chance to change this "stat_error" ? so it is consistent with https://github.com/ceph/ceph/pull/8983/files#diff-c24acfa82f15c38ae9370cab8630a579R1267 ?

File: src/tools/rados/rados.cc [hunk 2]
```diff
@@ -1317,20 +1325,26 @@ static void dump_inconsistent(const inconsistent_obj_t& inc,
...
+
+  f.open_array_section("errors");
+  if (inc.has_attr_unexpected())
+    f.dump_string("error", "attr_unexpected");
   if (inc.has_shard_missing())
-    f.dump_bool("missing", inc.has_shard_missing());
+    f.dump_string("error", "missing");
   if (inc.has_stat_error())
-    f.dump_bool("stat_err", inc.has_stat_error());
+    f.dump_string("error", "stat_err");
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: and change "stat_err" to "stat_error", so it is consistent with "read_error" below.

File: src/tools/rados/rados.cc [hunk 3]
```diff
@@ -1246,33 +1246,45 @@ static void dump_shard(const shard_info_t& shard,
...
+    f.dump_format("omap_digest", "0x%08x", shard.omap_digest);
+  }
+  if (shard.data_digest_present) {
+    f.dump_format("data_digest", "0x%08x", shard.data_digest);
+  }
+
+  f.open_array_section("errors");
   if (shard.has_read_error())
-    f.dump_bool("read_error", shard.has_read_error());
+    f.dump_string("error", "read_error");
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: could you add the handling for `shard.has_stat_error()` also?

**Comment by tchaikov** [comment]
@dzafman i am good with this new format. are we expecting some other opinions?

**Comment by dzafman** [comment]
Sample of what the output changes.

Previous format:
<pre> 
        {
            "name": "obj5",
            "nspace": "",
            "locator": "",
            "snap": "0x00000007",
            "oi_attr_missing": true,
            "headless": true
        },
        {
            "name": "obj2",
            "nspace": "",
            "locator": "",
            "snap": "snapdir",
            "ss_attr_missing": true,
            "extra_clones": true,
            "extra clones": [
                7,
                4
            ]
        },
</pre>

New format:
Notice the snap id is just a decimal value.  The errors are an array of strings instead of each flag shown as true.
<pre>
        {
            "name": "obj5",
            "nspace": "",
            "locator": "",
            "snap": 7,
            "errors": [
                "oi_attr_missing",
                "headless"
            ]
        },
       {
            "name": "obj2",
            "nspace": "",
            "locator": "",
            "snap": "snapdir",
            "errors": [
                "ss_attr_missing",
                "extra_clones"
            ],
            "extra clones": [
                7,
                4
            ]
        },
</pre>

**Comment by dzafman** [comment]
Requires ceph/ceph-qa-suite#1007 to be merged

**Pull request merged**